spec-main/extensions-spec.ts

```diff
@@ -193,9 +208,108 @@ describe('chrome extensions', () => {
     });
   });
 
+  describe('chrome.webRequest', () => {
+    function fetch (contents: WebContents, url: string) {
+      return contents.executeJavaScript(`fetch(${JSON.stringify(url)})`);
+    }
+
+    describe('onBeforeRequest', () => {
+      it('can cancel http requests', async () => {
+        const customSession = session.fromPartition(`persist:${uuid.v4()}`);
+        const w = new BrowserWindow({ show: false, webPreferences: { session: customSession, sandbox: true } });
+
+        w.loadURL(url);
+        await emittedOnce(w.webContents, 'dom-ready');
+        await customSession.loadExtension(path.join(fixtures, 'extensions', 'chrome-webRequest'));
+        await expect(fetch(w.webContents, url)).to.eventually.be.rejectedWith(TypeError);
+      });
+
+      it('does not cancel http requests when no extension loaded', async () => {
+        const customSession = session.fromPartition(`persist:${uuid.v4()}`);
+        const w = new BrowserWindow({ show: false, webPreferences: { session: customSession, sandbox: true } });
+
+        w.loadURL(url);
+        await emittedOnce(w.webContents, 'dom-ready');
+        await expect(fetch(w.webContents, url)).to.not.be.rejectedWith(TypeError);
+      });
+    });
+
+    it('takes precedence over Electron webRequest - http', async () => {
+      const customSession = session.fromPartition(`persist:${uuid.v4()}`);
+      const w = new BrowserWindow({ show: false, webPreferences: { session: customSession, sandbox: true } });
+
+      w.loadURL(url);
+      await emittedOnce(w.webContents, 'dom-ready');
+      await customSession.loadExtension(path.join(fixtures, 'extensions', 'chrome-webRequest'));
+      customSession.webRequest.onBeforeRequest(() => {
+        throw new Error('Electron onBeforeRequest callback has been called');
+      });
+      await expect(fetch(w.webContents, url)).to.eventually.be.rejectedWith(TypeError);
+    });
+
+    it('does not take precedence over Electron webRequest with different filter - http', async () => {
+      const customSession = session.fromPartition(`persist:${uuid.v4()}`);
+      const w = new BrowserWindow({ show: false, webPreferences: { session: customSession, sandbox: true } });
+
+      w.loadURL(url);
+      await emittedOnce(w.webContents, 'dom-ready');
+      await customSession.loadExtension(path.join(fixtures, 'extensions', 'chrome-webRequest'));
+      customSession.webRequest.onBeforeRequest({ urls: ['*://*.google.com/*'] }, async (details, callback) => {
+        callback({ cancel: true });
+      });
+      await expect(fetch(w.webContents, 'https://google.com')).to.eventually.be.rejectedWith(TypeError);
```

Comment:
Can we change these to `https://does.not.exist` or `https://127.127.127.127` some other obviously nonsense URL? Our tests should never make external requests.